crates/ruff_python_parser/src/lexer.rs

```diff
@@ -304,14 +303,18 @@ impl<'source> Lexer<'source> {
 
         let is_float = match self.cursor.rest().as_bytes() {
             [b'e' | b'E', b'0'..=b'9', ..] | [b'e' | b'E', b'-' | b'+', b'0'..=b'9', ..] => {
-                value_text.push('e');
-                self.cursor.bump(); // e | E
+                if self.cursor.bump() == Some('e') {
+                    number.push_matching('e');
+                } else {
+                    // 'E'
+                    number.push_different('e');
+                }
```

Comment:
Is there a specific reason that we're normalizing `E`? As per the [documentation of `f64::from_str`](https://doc.rust-lang.org/stable/std/primitive.f64.html#method.from_str), it accepts both `e` and `E`. I've tested it out locally as well with combinations of `(+|-)(e|E)`.